include/Queen.h

```diff
@@ -0,0 +1,37 @@
+#pragma once
+#include "Piece.h"
+
+
+/**
+ * @brief Represents a queen chess piece.
+ */
+class Queen : public Piece {
+public:
+	Queen(int color, Board* board, int row, int col);
+	bool isValidMove(int target_row, int target_col) override;
+	std::vector<std::tuple<int, int>> getAllvalidMoves() override;
+private:
+
+	/**
+	 * @brief helper function -  finds all valid moves for queen on diagonal runing from top left to bottom right of board
+	 * @param moves vector of already found legal moves
+	**/
+	void traverseleftToRightDiagonal(std::vector<std::tuple<int, int>>& moves);
```

Comment:
these functions are also declared in other classes, consider adding Intermediate classes for walking diagonally and walking straight, thus giving a finer inheritance for different tools.

Reply:
But then I will Have an issue with the Queen that will cause diamond inheritance.